packages/cells/src/widget.ts

```diff
@@ -1603,6 +1603,14 @@ export class CodeCell extends Cell<ICodeCellModel> {
     }
     // This is to hide/show icon on single line output.
     this.updatePromptOverlayIcon();
+
+    // Clear output area when empty
+    if (this.model.outputs.length === 0) {
+      this._lastOutputHeight = this.outputArea.node.style.height;
```

Comment:
I tested it locally and it was still not preserving the height when re-running a cell because this signal is emitted more than once. Also, it would force the previous height even if it was changed. I will push a commit to fix it.

Reply:
https://github.com/jupyterlab/jupyterlab/pull/16208/commits/50eda95fbb27ac52a72ca5b8feb054f1de97704a